main-app/src/main/java/com/nyctransittracker/mainapp/service/StopService.java

```diff
@@ -10,15 +10,17 @@
 public class StopService {
 
     private final StopRepository repository;
-    public void createStation(Stop stop, String direction, String route) {
-        var newStop = new Stop();
-        newStop.setUniqueId(stop.getGTFSStopID().toString() + route + direction);
-        newStop.setGTFSStopID(stop.getGTFSStopID());
-        newStop.setName(stop.getName());
-        newStop.setLatitude(stop.getLatitude());
-        newStop.setLongitude(stop.getLongitude());
-        newStop.setDirection(direction);
-        newStop.setRoute(route);
-        repository.save(newStop);
+    public void createStop(Stop stop, String direction, String route) {
+        if (!repository.findById(route).isPresent()) {
```

Comment:
is findById(route) ever found? 
Stop unique Id is comprised of (stop.getGTFSStopID() + route + direction) in line 16.